src/util/path.js

```diff
@@ -507,7 +507,7 @@
         p, nextLen, nextStep = 0.01, angleFinder = segInfo.angleFinder, lastPerc;
     // nextStep > 0.0001 covers 0.00015625 that 1/64th of 1/100
     // the path
-    while (tmpLen < distance && perc <= 1 && nextStep > 0.0001) {
+    while (tmpLen < distance && perc < (perc + nextStep) && nextStep > 0.0001) {
```

Comment:
mmm this smells like infinite loop

Reply:
can be this just < ( 1 + nextStep ), would this work?

Reply:
yeah. I think 1 + nextStep is the correct one.

Reply:
I tried, 1 + nextStep does not work. I think we can remove the percentage check as `tmpLen < distance` is enough.

Reply:
I think we can remove the percentage check. Only `tmpLen < distance && nextStep > 0.0001` are enough